dask/async.py

```diff
@@ -508,15 +499,16 @@ def fire_task():
             while state['ready'] and len(state['running']) < num_workers:
                 fire_task()
 
-    except KeyboardInterrupt:
-        for cb in started_cbs:
-            if cb[-1]:
-                cb[-1](dsk, state, True)
+    except BaseException:
+        for _, _, _, _, finish in started_cbs:
+            if finish:
+                finish(dsk, state, True)
         raise
 
     # Final reporting
+    # XXX: this may not be needed anymore
```

Comment:
@mrocklin, I think this check may no longer be needed. The previous `queue.get()` below had the wrong number of output args, indicating that it's never hit. I think the change is due to the addition of `cull` a while back, meaning this case is always `False` (I may be wrong on the diagnosis though).